src/Settings/View/AbstractView.php

```diff
@@ -4,51 +4,85 @@
+use MyParcelNL\Pdk\Base\Support\Arrayable;
+use MyParcelNL\Pdk\Base\Support\Collection;
 use MyParcelNL\Pdk\Base\Support\Utils;
-use MyParcelNL\Pdk\Form\Model\Input\CheckboxInput;
-use MyParcelNL\Pdk\Form\Model\Input\HiddenInput;
-use MyParcelNL\Pdk\Form\Model\Input\RadioButtonInput;
-use MyParcelNL\Pdk\Form\Model\Input\Select\CountrySelectInput;
-use MyParcelNL\Pdk\Form\Model\Input\Select\DropOffDaySelect;
-use MyParcelNL\Pdk\Form\Model\Input\SelectInput;
-use MyParcelNL\Pdk\Form\Model\Input\TextInput;
-use MyParcelNL\Pdk\Form\Model\Input\ToggleInput;
-
-class AbstractView
+use MyParcelNL\Pdk\Facade\DefaultLogger;
+
+abstract class AbstractView implements Arrayable
 {
-    protected const INPUT_LIST = [
-        Constant::TEXT                => TextInput::class,
-        Constant::TOGGLE              => ToggleInput::class,
-        Constant::SELECT              => SelectInput::class,
-        Constant::CHECKBOX            => CheckboxInput::class,
-        Constant::RADIO               => RadioButtonInput::class,
-        Constant::HIDDEN              => HiddenInput::class,
-        Constant::COUNTRY_SELECT      => CountrySelectInput::class,
-        Constant::DROP_OFF_DAY_SELECT => DropOffDaySelect::class,
-    ];
+    protected $data;
+
+    /**
+     * @var
+     */
+    protected $fields;
+
+    /**
+     * @param $data
+     */
+    public function __construct($data)
+    {
+        if (isset($data)) {
+            $this->data = $data;
+        }
+    }
 
     /**
-     * @var array
+     * @return \MyParcelNL\Pdk\Base\Support\Collection
      */
-    protected $attributes = [];
+    abstract protected function getFields(): Collection;
 
-    public function __construct(?array $data = null)
+    /**
+     * @return array
+     */
+    public function toArray(): array
     {
-        $this->attributes = $data['attributes'];
+        if (! isset($this->fields)) {
+            $this->fields = $this->getFields();
+
+            if (! empty($this->data)) {
+                $this->fields->push($this->data);
+            }
+        }
+
+        return $this
+            ->fields
+            ->map(function ($item) {
+                $validate = $this->validate($item);
+                if ($validate) {
+                    $item['type'] = Utils::classBasename($item['class']);
+                    unset($item['class']);
+                }
+
+                return $item;
+            })
+            ->toArray();
     }
 
     /**
-     * @param $input
+     * @param $item
      *
-     * @return mixed|string
+     * @return bool
      */
-    public function getClass($input)
+    private function validate($item): bool
     {
-        if (array_key_exists($input, self::INPUT_LIST)) {
-            return Utils::classBasename(self::INPUT_LIST[$input]);
+        if (isset($item['type'])) {
+            unset($item['type']);
+        }
+
+        if (! $item['name'] || ! $item['label']) {
+            DefaultLogger::warning(
```

Comment:
exception throwen, dit mag gewoon niet. dit gaat de eindgebruiker nooit zien, en wij gaan logger errors zelf ook nooit zien. Dit is puur voor wanneer wij de views aanpassen